src/redux/selectors/index.js

```diff
@@ -196,16 +178,140 @@ export const patientAgeAtResourcesSelector = createSelector(
+    return false;
+  }
+  if (!dateRangeStart || !dateRangeEnd) { return true; }
+  return (
+    isWithinInterval(
+      timelineDate, { start: startOfDay(dateRangeStart), end: endOfDay(dateRangeEnd) },
+    )
+  );
+};
+
+const filteredResourceTypesSelector = createSelector(
+  [
+    resourceTypeFiltersSelector,
+    resourceIdsGroupedByTypeSelector,
+    selectedResourceTypeSelector,
+    selectedCollectionResourceIdsSelector,
+    dateRangeFilterFiltersSelector,
+    resourcesSelector,
+    timelinePropsSelector,
+  ],
+  (
+    resourceTypeFilter,
+    resourceIdsGroupedByType,
+    selectedResourceType,
+    selectedCollectionResourceIdsObjects,
+    dateRangeFilterFilters,
+    resources,
+    timelineProps,
+  ) => {
+    if (!selectedCollectionResourceIdsObjects) {
+      return {};
+    }
+    const { minimumDate, maximumDate } = timelineProps;
+    const { dateRangeStart, dateRangeEnd } = dateRangeFilterFilters;
+    const activeDateStart = dateRangeStart || minimumDate;
+    const activeDateEnd = dateRangeEnd || maximumDate;
+    const selectedCollectionResourceIds = Object.keys(selectedCollectionResourceIdsObjects);
+    return Object.keys(resourceTypeFilter).reduce((acc, resourceType) => {
+      if (resourceTypeFilter[resourceType]) {
+        acc[resourceType] = {};
+        acc[resourceType].selected = false;
+        if (selectedResourceType === resourceType) {
+          acc[resourceType].selected = true;
+        }
+        Object.entries(resourceIdsGroupedByType[resourceType]).forEach(([subType, resourceIds]) => {
+          if (!acc[resourceType].subTypes) {
+            acc[resourceType].subTypes = {};
+          }
+          if (!acc[resourceType].subTypes[subType]) {
+            acc[resourceType].subTypes[subType] = {};
+          }
+          const subTypeResourceIds = Array.from(resourceIds);
+          acc[resourceType].subTypes[subType].resourceIds = subTypeResourceIds;
+          acc[resourceType].subTypes[subType].count = subTypeResourceIds.length;
+          const dateFilteredResourceIds = subTypeResourceIds
+            .filter((subTypeResourceId) => isResourceInDateRange(
+              resources[subTypeResourceId], activeDateStart, activeDateEnd,
+            ));
+          acc[resourceType].subTypes[subType].dateFilteredResourceIds = dateFilteredResourceIds;
+          acc[resourceType].subTypes[subType].dateFilteredCount = dateFilteredResourceIds.length;
+          const collectionDateFilteredResourceIds = dateFilteredResourceIds
+            .filter((dateFilteredResourceId) => selectedCollectionResourceIds
+              .includes(dateFilteredResourceId));
+          acc[resourceType].subTypes[subType]
+            .collectionDateFilteredResourceIds = collectionDateFilteredResourceIds;
+          acc[resourceType].subTypes[subType]
+            .collectionDateFilteredCount = collectionDateFilteredResourceIds.length;
+        });
```

Comment:
I wonder if it could be more clear w/ an intervening object assignment, and/or Object literal notation

```
const subtypeViewModel = acc[resourceType].subTypes[subType]

const subtypeViewModel = {
...subtypeViewModel,
  foo: bar, 
  baz: biff
}
```

Reply:
b/c grouping requires an intervening representation as a hash, it might be simpler to sort (by label) in the UI, and only for expanded accordions.